packages/components/src/heading/hook.ts

```diff
@@ -56,16 +56,17 @@ export function useHeading(
 		'Heading'
 	);
 
-	const as = asProp || `h${ level }`;
+	const as = ( asProp || `h${ level }` ) as keyof JSX.IntrinsicElements;
```

Comment:
Necessary to avoid a TypeScript error when passing this computed `as` prop to the `View` component